src/fheroes2/gui/interface_events.cpp

```diff
@@ -189,6 +189,10 @@ fheroes2::GameMode Interface::AdventureMap::EventHeroMovement()
         else if ( MP2::isInGameActionObject( hero->getObjectTypeUnderHero(), hero->isShipMaster() ) ) {
             return EventDefaultAction();
         }
+        else if ( !hero->MayStillMove( false, true ) ) {
```

Comment:
I recommend to follow the logic defined in the `Interface::ButtonsPanel::_setButtonStatus()`:

```suggestion
        else if ( hero->GetPath().isValidForMovement() ) {
```

Otherwise, this message will be shown even if hero has no predefined path at all (if player presses the corresponding hotkey):

https://github.com/user-attachments/assets/68be1028-7af4-4b31-a7fe-1aef0931092d

Reply:
I think it won't be a big deal if this message is also displayed when the hotkey is used.